packages/valory/skills/termination_abci/rounds.py

```diff
@@ -0,0 +1,126 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the termination round classes."""
+from enum import Enum
+from typing import Optional, Tuple
+
+from packages.valory.skills.abstract_round_abci.abci_app_chain import chain
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbstractRound,
+    BaseSynchronizedData,
+    BaseTxPayload,
+    CollectSameUntilThresholdRound,
+)
+from packages.valory.skills.termination_abci.payloads import BackgroundPayload
+from packages.valory.skills.transaction_settlement_abci.rounds import (
+    FailedRound,
+    FinishedTransactionSubmissionRound,
+    TransactionSubmissionAbciApp,
+)
+
+
+class Event(Enum):
+    """Defines the (background) round events."""
+
+    TERMINATE = "terminate"
+
+
+class SynchronizedData(BaseSynchronizedData):
+    """
+    Class to represent the synchronized data.
+
+    This data is replicated by the tendermint application.
+    """
+
+    @property
+    def safe_contract_address(self) -> Optional[str]:
+        """Get the safe contract address."""
+        return cast(Optional[str], self.db.get("safe_contract_address", None))
+
+    @property
+    def termination_majority_reached(self) -> bool:
+        """Get termination_majority_reached."""
+        return cast(bool, self.db.get("termination_majority_reached", False))
+
+
+class BackgroundRound(CollectSameUntilThresholdRound):
+    """Defines the background round, which runs concurrently with other rounds."""
+
+    round_id: str = "background_round"
+    allowed_tx_type = BackgroundPayload.transaction_type
+    payload_attribute: str = "background_data"
+    synchronized_data_class = SynchronizedData
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            state = self.synchronized_data.update(
+                synchronized_data_class=self.synchronized_data_class,
+                termination_majority_reached=True,
+                most_voted_tx_hash=self.most_voted_payload,
+            )
+            return state, Event.TERMINATE
+
+        return None
+
+
+class TerminationRound(AbstractRound):
+    """Round to act as the counterpart of the behaviour responsible for terminating the agent."""
+
+    round_id = "termination_round"
+
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
+
+    def process_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """No logic required here."""
+        return None
```

Comment:
Why does this have no implementation?

Reply:
Because we don't have any logic here, the moment the agent enters this behaviour it gets terminated.